### PR TITLE
Added functionality to log smiles in smi format

### LINKED ISSUE #25: Log SMILES string for which the generation of the structure files fails
recetox-msfinder does not produce structure files for the SMILES, representing a mixture or a coordination complex. We can check if the structure result object is null in the PeakAssigner class process function and write a log file which outputs the smiles for which the generation of the structure files fails.

### PR DESCRIPTION
**Pull request by zargham-ahmad**
This PR include the following:

- added functionality to log smile in smi format for those spectra for which structure files are not produced.
- added a message to the console if the count of data, formulaResults, and sfdResults are not equal.
- added test and testdata

closes #25

### CODE REVIEW

File: MsfinderCommon/Utility/FileStorageUtility.cs [hunk 1]
```diff
@@ -473,20 +473,48 @@ public sealed class FileStorageUtility {
...
-        public static void DeleteSfdFiles(string[] structureFiles) {
-            foreach (var file in structureFiles) {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return folderPath + "\\" + "log_smiles.smi";
+            }
+            else
+            {
+                return folderPath + "/" + "log_smiles.smi";
+            }
```
Comment (hechth) [inline MsfinderCommon/Utility/FileStorageUtility.cs:487]: Isn't there some built-in function which can handle paths in a platform independent way?

File: MsfinderCommon/Utility/FileStorageUtility.cs [hunk 2]
```diff
@@ -473,20 +473,48 @@ public sealed class FileStorageUtility {
...
+        }
+        public static void DeleteSfdFiles(string[] structureFiles)
```
Comment (hechth) [inline MsfinderCommon/Utility/FileStorageUtility.cs:489]: I think they are called sdf and not sfd

File: MsfinderCommon/Process/PeakAssigner.cs [hunk 3]
```diff
@@ -44,11 +44,17 @@ public sealed class PeakAssigner
...
             foreach (var result in results) {
                 result.TotalScore += formulaResult.TotalScore;
                 result.Ontology = rawData.Ontology;
             }
+            if (results.Count == 0)
+            {
+            // If the results is 0, write to a file in SMI format
+                var logSmileFilePath = FileStorageUtility.GetSmileLogFilePath(analysisFile.FormulaFilePath);
+                FileStorageUtility.WriteToSmiFile(rawData.Smiles, logSmileFilePath);
+            }
```
Comment (hechth) [inline MsfinderCommon/Process/PeakAssigner.cs:57]: So this means overall if the fragmenter didn't find any results? Is that also the case if everything is parsed correctly, just there are no peak annotations found?

**Review by hechth — COMMENTED** [review]
Would be great to have a unit test for this - to have a SMILES which the tool can't handle and then grab the log file to see that it is there.

File: MsfinderCommon/Utility/FileStorageUtility.cs (hunk 2 above)
Comment (zargham-ahmad) [reply MsfinderCommon/Utility/FileStorageUtility.cs]: made in more generic in b37f876

File: MsfinderCommon/Utility/FileStorageUtility.cs (hunk 1 above)
Comment (zargham-ahmad) [reply MsfinderCommon/Utility/FileStorageUtility.cs]: updated in b37f876

**Comment by zargham-ahmad** [comment]
> Would be great to have a unit test for this - to have a SMILES which the tool can't handle and then grab the log file to see that it is there.

added test in b37f876

**Review by hechth — APPROVED** [review]

**Pull request merged**